app/gui/view/graph-editor/src/component/node/input/widget/method.rs

```diff
@@ -66,14 +79,42 @@ impl SpanWidget for Widget {
         display_object
             .use_auto_layout()
             .set_row_flow()
-            .set_gap_x(ICON_GAP)
             .set_children_alignment_left_center()
             .justify_content_center_y();
-
         let icon_wrapper = object::Instance::new_named("icon_wrapper");
         icon_wrapper.set_size((SIZE, SIZE));
-        let mut this = Self { display_object, icon_wrapper, icon_id: default(), icon: default() };
-        this.set_icon(IconId::default());
+        let icon_view = Rc::new(RefCell::new(AnyIcon::default()));
+
+        let frp = Frp::new();
+        let network = &frp.network;
+        let style = ctx.cached_style::<Style>(network);
+        frp::extend! { network
+            icon_gap <- style.on_change().map(|style| style.icon_gap);
+            eval icon_gap((gap) display_object.set_gap_x(*gap).void());
+
+            set_icon <- frp.set_icon.on_change();
+            eval set_icon([icon_view, icon_wrapper] (icon_id) {
+                let icon = icon_id.cached_view();
+                icon.set_size((SIZE, SIZE));
+                icon_wrapper.replace_children(&[icon.display_object()]);
+                icon_view.replace(icon);
+            });
+
+            icon_state_dirty <- any(...);
+            icon_state_dirty <+ frp.set_icon_state.on_change();
+            icon_state_dirty <+ frp.set_icon_state.sample(&set_icon);
+            icon_alpha <- all_with(&icon_state_dirty, &style, |state, style| state.alpha(style));
+            eval icon_alpha((alpha)
+                icon_view.borrow().r_component.set(Vector4(1.0, 1.0, 1.0, *alpha)));
+        }
+        let this = Self {
+            display_object,
+            icon_wrapper,
+            icon_view,
+            frp,
+        };
+        this.frp.set_icon(IconId::default());
+        this.frp.set_icon_state(IconState::Ready);
```

Comment:
I don't think we need the `this` variable here. The initialization could be called on local `frp` directly, then the constructed `Self` can be returned immediately.